### PR TITLE
feature: implemented the cache:get_keys method.

### PR DESCRIPTION
**Pull request by thibaultcha**
Built on top of #25 for the `self.num_items` attribute.

### CODE REVIEW

File: README.markdown [hunk 1]
```diff
@@ -180,6 +183,45 @@ Removes an item specified by the key from the cache.
 
 [Back to TOC](#table-of-contents)
 
+count
+-----
+`syntax: count = cache:count()`
+
+Returns the number of items currently inside the cache, **including** expired
+items, if any.
```
Comment (thibaultcha) [inline README.markdown]: should specify: in their LRU order

File: README.markdown [hunk 2]
```diff
@@ -180,6 +183,45 @@ Removes an item specified by the key from the cache.
...
+Returns the maximum number of items this cache can hold. This is the same as
+the `size` parameter passed to [`cache:new`](#new) when the cache was created.
+
+[Back to TOC](#table-of-contents)
+
+get_keys
+--------
+`syntax: keys = cache:get_keys(max_count?, res?)`
+
+Fetch the list of keys currently inside the cache up to `max_count`.
```
Comment (thibaultcha) [inline README.markdown]: Should specify that the retrieved keys will be in MRU order.

**Comment by agentzh** [comment]
@thibaultcha Not a priority for me atm, but I'll definitely have a look. Thank you and sorry for the delay on my side!

**Comment by thibaultcha** [comment]
@dndx Thanks! I rebased it on top of your branch.

**Comment by valeriano-manassero** [comment]
Hi,
when do you think this PR will be merged?
Thank you.

**Comment by thibaultcha** [comment]
@agentzh I'm afraid that since this PR is based on top of #25, we first need #25 to be rebased (or merged) before I can do so here (cc. @dndx)

File: README.markdown [hunk 3]
```diff
@@ -180,6 +183,46 @@ Removes an item specified by the key from the cache.
...
+`count` is always less than or equal to `size` as passed to [`cache:new`](#new)
+and greater than or equal to 0.
+
+This is a good way of telling of how much memory the cache is actually consuming.
+
+[Back to TOC](#table-of-contents)
+
+capacity
+--------
+`syntax: cap = cache:capacity()`
```
Comment (agentzh) [inline README.markdown]: As mentioned in the original PR's review, should use the variable name `count` here instead of `cap`.

**Comment by thibaultcha** [comment]
Merged!

**Pull request closed without merging**